src/components/Firestore/testing/FirestoreTestProviders.tsx

```diff
@@ -0,0 +1,97 @@
+import * as firebaseTesting from '@firebase/testing';
+import { render, waitForElement } from '@testing-library/react';
+import * as firebase from 'firebase';
+import React, { useEffect, useState } from 'react';
+import { Provider } from 'react-redux';
+import { MemoryRouter } from 'react-router-dom';
+import { FirebaseAppProvider, preloadFirestore, useFirestore } from 'reactfire';
+
+import configureStore from '../../../configureStore';
+import { fetchSuccess } from '../../../store/config/actions';
+
+interface RenderOptions {
+  path?: string;
+}
+
+export const renderWithFirestore = async (
+  children: (
+    firestore: firebase.firestore.Firestore
+  ) => Promise<React.ReactElement>,
+  options: RenderOptions = {}
+) => {
+  const component = render(
+    <FirestoreTestProviders path={options.path}>
+      <AsyncFirestore>{children}</AsyncFirestore>
+    </FirestoreTestProviders>
+  );
+
+  await waitForElement(() => component.findByTestId('firestore-foo'));
+
+  return component;
+};
+
+export const FirestoreTestProviders: React.FC<RenderOptions> = React.memo(
+  ({ children, path = '' }) => {
+    const projectId = `${process.env.GCLOUD_PROJECT}-${Date.now()}`;
+    const hostAndPort =
+      process.env.FIRESTORE_EMULATOR_HOST ||
+      process.env.FIREBASE_FIRESTORE_EMULATOR_ADDRESS;
+    if (!projectId || !hostAndPort) {
+      throw new Error('FirestoreTestProviders requires a running Emulator');
+    }
+    const [host, port] = hostAndPort.split(':');
+
+    const store = configureStore();
+    store.dispatch(
+      fetchSuccess({
+        projectId,
+        firestore: {
+          hostAndPort,
+          host,
+          port: Number(port),
+        },
+      })
+    );
+
+    const testingApp = firebaseTesting.initializeTestApp({
+      projectId,
+    });
+
+    useEffect(() => {
+      return () => {
+        testingApp.delete();
+      };
+    }, []);
+
+    return (
+      <Provider store={store}>
```

Comment:
IMHO this should just use a fake store. The configureStore and dispatch above is way too much effort.

Reply:
Updated with redux-mock-store